lib/config/rules/variables.js

```diff
@@ -20,7 +20,7 @@ module.exports = {
   // Disallow use of undeclared variables unless mentioned in a /*global */ block
   'no-undef': 'error',
   // Disallow use of undefined variable
-  'no-undefined': 'error',
+  'no-undefined': 'off',
```

Comment:
I think this is a kind of antiquated rule. It is also very hard to follow in TS when we often type things as being optional for which `undefined` is the only allowable value to "unset" the field.